Show most recent hang stacks

Uberi opened the pull request:
* Up to 10 of the most recent hang stacks are displayed in the panel, from most recent to least recent.
* Stacks can be copied/pasted and the list updates as new stacks come in.

It looks like this:

![screenshot from 2016-01-18 14-07-11](https://cloud.githubusercontent.com/assets/437196/12400384/adb971c6-bded-11e5-8b11-cf3184e9b4ff.png)

Uberi commented:
Also, this fixes #11.

chutten commented:
why not `git rebase -i` to pull that style fix after the merge/rebase so you're only submitting what you ought to be submitting?

Uberi commented:
Rebased off gh-pages and merged in the changes from `sounds`.

chutten commented on data/panel.js:
```diff
@@ -51,3 +51,29 @@ self.port.on("warning", function(warningType) {
+self.port.on("set-hangs", function(hangStacks) {
+  var hangs = document.getElementById("hangStacks");
+  hangs.innerHTML = ""; // clear the contents of the hang stacks list
+  hangStacks.reverse().forEach((hangStack, i) => {
```
you don't use `i`, do you? You can just have `hangStack => {`

chutten commented on index.js:
```diff
@@ -184,6 +191,24 @@ function numEventLoopLags() {
+  if (geckoThread.hangs.length == 0) { // No hangs found
+    return [];
+  }
+  var recentHangs = geckoThread.hangs.slice(Math.max(0, geckoThread.hangs.length - 10));
```
Could you not just write as `var recentHangs = geckoThread.hangs.slice(-10);`

chutten commented on index.js:
```diff
@@ -184,6 +191,24 @@ function numEventLoopLags() {
+    return [];
+  }
+  var recentHangs = geckoThread.hangs.slice(Math.max(0, geckoThread.hangs.length - 10));
+  return recentHangs.map((hangEntry, i) => {
```
ditto not using `i`

chutten commented on index.js:
```diff
@@ -184,6 +191,24 @@ function numEventLoopLags() {
+  }
+  var recentHangs = geckoThread.hangs.slice(Math.max(0, geckoThread.hangs.length - 10));
+  return recentHangs.map((hangEntry, i) => {
+    return hangEntry.stack.slice(0).reverse().join("\n");
```
not sure you need the `slice(0)` for the shallow copy since you already got a shallow copy from the `slice(-10)` in line 206

Uberi replied:
This is shallow copying the `stack` property of each `hangEntry`, which is the original reference since the `slice` on the array of `hangEntry` only does a shallow copy. However, if it was a deep copy on line 206, this would be redundant.

chutten replied:
Ah, yes, of course. Comment retracted.

Uberi commented:
Rebased off gh-pages again, but also implemented :vladan's suggested changes to the feed. It now looks like this:

![screenshot from 2016-01-19 13-46-31](https://cloud.githubusercontent.com/assets/437196/12428882/34fc5a00-beb5-11e5-8bbb-d1bc6167717f.png)

I think I also got all of your code review suggestions in as well, let me know if there's anything else.

chutten commented on data/panel.html:
```diff
@@ -58,5 +101,25 @@
+  <h1>MOST RECENT HANGS</h1>
+  <div id="hangStacks">
+    <!-- hang stack entries are added here -->
+    <div>
```
You probably don't want to leave this fake UI in, just in case something explodes and lets the user see this.

chutten commented:
How well does this behave on a debug build (with BHR off)?

Uberi commented:
- I turned the fake entry into a comment (it's there to make it easy to see what the structure of the entries will be for styling and other purposes).
- When BHR is off, you get a `?` badge for the icon, a big orange banner explaining the problem in the panel, and the hang stacks list says, "No hang data available.". I just added a commit that displays the message upon first opening.
